apps/webapp/src/state/helpers.ts

```diff
@@ -1,18 +1,54 @@
 import { TransactionPlannerRequest } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1alpha1/view_pb';
 import { viewClient } from '../clients/grpc.ts';
 import { uint8ArrayToHex } from '@penumbra-zone/types';
+import { TransactionId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/txhash/v1alpha1/txhash_pb';
+import { sha256Hash } from '@penumbra-zone/crypto-web';
 
-export const planWitnessBuildBroadcast = async (req: TransactionPlannerRequest) => {
-  const { plan } = await viewClient.transactionPlanner(req);
-  if (!plan) throw new Error('no plan in response');
+export const planWitnessBuildBroadcast = async (
+  req: TransactionPlannerRequest,
+  awaitDetection = true,
```

Comment:
```suggestion
```
We're not actually passing this argument in anywhere — OK if we just hard-code it as `true` in the call to `viewClient.broadcastTransaction` on line 30 below?

Reply:
this is a helper method now and not local to any specific slice, let's allow it to remain versatile